File: internal/infra/di/check_payment_status.go
package di

import (
	driven_dynamodb "github.com/fiap/ms-payment/internal/adapters/driven/dynamodb"
	"github.com/fiap/ms-payment/internal/core/application/usecases"
	infra_aws "github.com/fiap/ms-payment/internal/infra/aws"
)

type CheckPaymentStatusContainer struct {
	CheckPaymentStatusUseCase *usecases.CheckPaymentStatusUseCase
}

func NewCheckPaymentStatusContainer(region infra_aws.AwsRegion, stage infra_aws.AwsStage) *CheckPaymentStatusContainer {
	dynamoClient := infra_aws.NewDynamoDBClient(region, stage)

	paymentRepo := driven_dynamodb.NewPaymentRepository(*dynamoClient)

	return &CheckPaymentStatusContainer{
		CheckPaymentStatusUseCase: usecases.NewCheckPaymentStatusUseCase(paymentRepo),
	}
}

File: src/infra/di/check_payment_status.ts
import { CheckPaymentStatusUseCase } from "@usecases/check_payment_status"
import { PaymentRepository } from "@driven_dynamodb/payment_repository"
import { DynamoDBClientWrapper } from "@aws/dynamodb_client"
import { AwsRegion, AwsStage } from "@aws/utils"

export class CheckPaymentStatusContainerFactory {
	usecase: CheckPaymentStatusUseCase

	constructor(region: AwsRegion, stage: AwsStage) {
		const dynamoClient = new DynamoDBClientWrapper(region, stage)
		const paymentRepo = new PaymentRepository(dynamoClient)

		this.usecase = new CheckPaymentStatusUseCase(paymentRepo)
	}
}